src/data/collections/mutable/MutableSet.ts

```diff
@@ -209,14 +209,17 @@ class MutableSet<T> extends BaseCollection<T> implements Collection<T> {
     
     exportMutableState() {
         return {
-            _elements: this._elements,
-            _currentAddOpRefs: this._currentAddOpRefs
+            _elements: Object.fromEntries(this._elements.entries()),
+            _currentAddOpRefs: Object.fromEntries([...this._currentAddOpRefs.entries()].map(([key, value]) => [key, value.literalize()]))
         };
     }
 
-    importMutableState(state: any): void {
+    importMutableState(state: {
+        _elements: {[key: string]: T},
+        _currentAddOpRefs: {[key: string]: {value: HashedSetLiteralized }}
+    }): void {
         this._elements = new Map(Object.entries(state._elements));
-        this._currentAddOpRefs = new Map(Object.entries(state._currentAddOpRefs));
+        this._currentAddOpRefs = new Map(Object.entries(state._currentAddOpRefs).map(([key, value]) => [key, HashedSet.deliteralize(value, new Context())]));
```

Comment:
Here's where I tried to deliteralize the `HashedSet`, with no luck

Reply:
There was a typing error that was not being catched by the type system, HashedSet's `literalize` was returning a `{value: value, dependencies: dependencies}`, and while restoring the checkpoint we were calling 

`HashedSet.deliteralize({value: value, dependencies: dependencies}, context)`

while the expected was

`HashedSet.deliteralize(value, context)`.

There's something we missed in our initial analysis: the literalization process will usually produce a map (hash -> literalization), that contains all the sub-objects of whatever was being literalized. This objects will be in the store, so we could

1) just save their hashes, and load them back from the store when applying the checkpoint (slower, but space-efficient)

2) save this in the checkpoint (faster, there will be some space duplication).

I adapted the code so that it would use 2).

Since the containers support storing both objects and plain literals, I had to check and have two code paths, for example here it is for `MutableReference`:

```typescript
    exportMutableState() {

        let objectValue  : (LiteralContext|undefined);
        let literalValue : any;

        if (this._value instanceof HashedObject) {
            const context = new Context();
            this._value.toContext(context);
            objectValue = context.toLiteralContext();
        } else {
            literalValue = this._value;
        }

        return {
            _sequence: this._sequence,
            _timestamp: this._timestamp,
            _objectValue: objectValue,
            _literalValue: literalValue
        };
    }

    importMutableState(state: any) {
        this._sequence = state._sequence;
        this._timestamp = state._timestamp;

        if (state._objectValue !== undefined) {
            this._value = HashedObject.fromLiteralContext(state._objectValue) as any as T;
        } else {
            this._value = state._literalValue;
        }
    }
```